rootfs/api/views.py

```diff
@@ -346,14 +346,38 @@ def get_object(self, **kwargs):
 
 
 class CertificateViewSet(BaseDeisViewSet):
-    """A viewset for interacting with Domain objects."""
+    """A viewset for interacting with Certificate objects."""
     model = models.Certificate
     serializer_class = serializers.CertificateSerializer
 
     def get_object(self, **kwargs):
-        """Retrieve domain certificate by common name"""
+        """Retrieve domain certificate by its name"""
         qs = self.get_queryset(**kwargs)
-        return qs.get(common_name=self.kwargs['common_name'])
+        return qs.get(name=self.kwargs['name'])
+
+    def attach(self, request, *args, **kwargs):
+        try:
+            # TODO how to validate domain is set in the body?
+            kwargs['domain'] = request.data['domain']
+            self.get_object().attach(*args, **kwargs)
+        except Http404:
+            raise
+        except AlreadyExists as e:
+            return Response({'detail': str(e)}, status=status.HTTP_409_CONFLICT)
+        except Exception as e:
+            return Response({'detail': str(e)}, status=status.HTTP_503_SERVICE_UNAVAILABLE)
```

Comment:
It was in line with the rest of the API error "handling" so I was mostly just keeping with convention. It can be a different return value if we figure out what's more appropriate for unrecoverable issue